packages/cloud_firestore/cloud_firestore_platform_interface/lib/src/settings.dart

```diff
@@ -44,13 +45,20 @@ class Settings {
   /// and can be set to [Settings.CACHE_SIZE_UNLIMITED] to disable garbage collection.
   final int? cacheSizeBytes;
 
+  /// Whether to skip nested properties that are set to undefined during object serialization.
+  /// If set to true, these properties are skipped and not written to Firestore. If set to false
+  /// or omitted, the SDK throws an exception when it encounters properties of type undefined.
+  /// Web only.
+  final bool ignoreUndefinedProperties;
+
   /// Returns the settings as a [Map]
   Map<String, dynamic> get asMap {
     return {
       'persistenceEnabled': persistenceEnabled,
       'host': host,
       'sslEnabled': sslEnabled,
-      'cacheSizeBytes': cacheSizeBytes
+      'cacheSizeBytes': cacheSizeBytes,
+      if (ignoreUndefinedProperties) 'ignoreUndefinedProperties': true,
```

Comment:
What is the reason to not do it like the other properties?
```dart
'ignoreUndefinedProperties': ignoreUndefinedProperties
```

Reply:
It's web only. My sense it would confuse users to keep it there when using Firestore for other platforms. Happy to reinstate if that's the consensus. Maybe even just show it if it is the web platform? Or just show it for all platforms.